src/shims/posix/foreign_items.rs

```diff
@@ -22,48 +22,54 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, bool> {
         let this = self.eval_context_mut();
 
-        check_abi(abi, Abi::C { unwind: false })?;
-
         match link_name {
             // Environment related shims
             "getenv" => {
+                check_abi(abi, Abi::C { unwind: false })?;
```

Comment:
Calling an `extern "C"` function as `extern "C-unwind"` would be fine I think as the only difference if whether unwinding is allowed or not.

Reply:
I think that's the result of #1739... I don't know if I need to change that here as well?

Reply:
So far Miri does not implement any kind of "ABI compatibility"; the ABIs have to match exactly. [The core interpreter engine does the same](https://github.com/rust-lang/rust/blob/0ce0fedb67fa66d50aa819ef8b12f1d89eb22d7d/compiler/rustc_mir/src/interpret/terminator.rs#L230).

I am aware that this is overly restrictive; however, I also don't think Rust is making any explicit promises about ABI compatibility yet. So even if e.g. `C` vs `C-unwind` *could* be fine, there's still the question of whether this is sanctioned by the Rust spec or not.